arangod/Graph/Cache/RefactoredTraverserCache.cpp

```diff
@@ -152,10 +160,30 @@ bool RefactoredTraverserCache::appendEdge(EdgeDocumentToken const& idToken,
                 // NOTE: Do not count this as Primary Index Scan, we
                 // counted it in the edge Index before copying...
                 if constexpr (std::is_same_v<ResultType, aql::AqlValue>) {
+                  if (!_edgeProjections.empty()) {
+                    // TODO: This does one unnecessary copy.
```

Comment:
I would like to keep this TODO.
This is a general optimization to allow constructing an AQL value out of slice & projection and could be used in other places as well (e.g. Index)